optimism/src/mips/interpreter.rs

```diff
@@ -784,8 +785,33 @@ pub fn interpret_rtype<Env: InterpreterEnv>(env: &mut Env, instr: RTypeInstructi
         RTypeInstruction::JumpAndLinkRegister => (),
         RTypeInstruction::SyscallMmap => (),
         RTypeInstruction::SyscallExitGroup => (),
+        RTypeInstruction::SyscallReadHint => (),
         RTypeInstruction::SyscallReadPreimage => (),
-        RTypeInstruction::SyscallReadOther => (),
+        RTypeInstruction::SyscallReadOther => {
+            let fd_id = env.read_register(&Env::constant(4));
+            let mut check_equal = |expected_fd_id: u32| {
+                // FIXME: Requires constraints
+                let pos = env.alloc_scratch();
+                unsafe { env.test_zero(&(fd_id.clone() - Env::constant(expected_fd_id)), pos) }
+            };
+            let is_stdin = check_equal(FD_STDIN);
+            let is_preimage_read = check_equal(FD_PREIMAGE_READ);
+            let is_hint_read = check_equal(FD_HINT_READ);
+
+            // FIXME: Should assert that `is_preimage_read` and `is_hint_read` cannot be true here.
```

Comment:
do you mean using `env.test_zero` or just a boolean check with normal asserts?

Reply:
I mean by adding a constraint, which is not exposed in the API yet.